packages/webos/LS2Request/LS2Request.js

```diff
@@ -54,14 +71,20 @@ export default class LS2Request {
 		this.bridge = new PalmServiceBridge();
 		this.bridge.onservicecallback = this.callback.bind(this, onSuccess, onFailure, onComplete);
 		this.bridge.call(adjustPath(service) + method, JSON.stringify(parameters));
-
+		if (timeout) {
+			this.timeoutJob.startAfter(timeout, {onTimeout, timeout});
+		}
 		return this;
 	}
 
 	callback (onSuccess, onFailure, onComplete, msg) {
 		if (this.cancelled) {
 			return;
 		}
+
+		// remove timeout job
+		this.timeoutJob.stop();
```

Comment:
This is a bit tangential but this line brought it to mind. It appears that we're not resetting the `cancelled` flag when sending a new request. As a result, if you cancel the first request, the second won't fire the callback nor cancel the timeout job (since it'll return early just before this line) causing `onTimeout` to fire for a successful second call.

@viodragon2 + @sugardave - Are you reading the same? If so, let's add that change here to proactively fix the bug I invented above.

Reply:
I think you're right, I'll fix it up.